src/views/footnote.ts

```diff
@@ -14,24 +14,115 @@
  * limitations under the License.
  */
 
-import { FootnoteNode, ManuscriptNode } from '@manuscripts/transform'
+import { FootnoteNode, schema } from '@manuscripts/transform'
+import { isEqual } from 'lodash'
+import { Transaction } from 'prosemirror-state'
+import { findParentNodeOfTypeClosestToPos } from 'prosemirror-utils'
 
-import { getChangeClasses } from '../lib/track-changes-utils'
-import BlockView from './block_view'
-import { createNodeOrElementView } from './creators'
+import {
+  DeleteFootnoteDialog,
+  DeleteFootnoteDialogProps,
+} from '../components/views/DeleteFootnoteDialog'
+import { alertIcon, deleteIcon } from '../icons'
+import { getFootnotesElementState } from '../lib/footnotes'
+import { isDeleted } from '../lib/track-changes-utils'
+import { BaseNodeView } from './base_node_view'
+import { createNodeView } from './creators'
+import ReactSubView from './ReactSubView'
 
-export class FootnoteView extends BlockView<FootnoteNode> {
-  public elementType = 'div'
-}
+export class FootnoteView extends BaseNodeView<FootnoteNode> {
+  dialog: HTMLElement
+
+  public initialise = () => {
+    this.dom = document.createElement('div')
+    this.dom.classList.add('footnote')
+    this.contentDOM = document.createElement('div')
+    this.contentDOM.classList.add('footnote-text')
+    this.updateContents()
+  }
+
+  public updateContents = () => {
+    const id = this.node.attrs.id
+    const fns = getFootnotesElementState(this.view.state, id)
+    if (!fns) {
+      return
+    }
+
+    const marker = document.createElement('span')
```

Comment:
am I missing something or is there no check and removal for these elements and their listeners created on previous calls of _updateContents_?

Reply:
ah, sorry. missed that. thanks